src/System.Net.Http/src/System/Net/Http/HttpClientHandler.Unix.cs

```diff
@@ -160,6 +161,10 @@ protected override void Dispose(bool disposing)
 
         protected internal override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         {
+            if (DiagnosticsHandler.IsEnabled())
```

Comment:
> However what is true is that if there are NO subscribers to the Listener, IsEnabled(string) returns very quickly (it checks a field for null and returns false).

@vancem, I was just looking at some ASP.NET traces, and IsEnabled(string) is actually showing up.  It's a virtual call, so it doesn't get inlined and involves virtual dispatch, and then even if it wasn't it's got enough IL to not be considered for inlining.  Seems like something that should be addressed.